recovery/elliptics_recovery/types/merge.py

```diff
@@ -702,12 +823,15 @@ def dump_process_group((ctx, group)):
                                  remotes=ctx.remotes)
     ret = True
     with open(ctx.dump_file, 'r') as dump:
+        ss_rec = ServerSendRecovery(ctx, node, group)
         # splits ids from dump file in batchs and recovers it
         for batch_id, batch in groupby(enumerate(dump), key=lambda x: x[0] / ctx.batch_size):
             recovers = []
             rs = RecoverStat()
-            for _, val in batch:
-                rec = DumpRecover(routes=ctx.routes, node=node, id=elliptics.Id(val), group=group, ctx=ctx)
+            keys = [elliptics.Id(val) for _, val in batch]
+            keys = ss_rec.recover(keys)
+            for k in keys:
```

Comment:
Server-send recovery will not work if servers in different groups are not connected, in particular, this happens when storage nodes are being firewalls in different physical locations and they are not connected via private network.